File: RetrofitTest/app/src/main/java/contract/FourthContract.java
package contract;

import info.LoginNetInfo;

public interface FourthContract {
    interface Model {

        void loginNet(String userName , String pwd , LoginNetInfo loginNetInfo);
    }

    interface View {

        String getUserName();
        String getPassWord();
        void loginSuccess(String success);
        void error(String error);

    }

    interface Presenter {
        //登录联网
        void loginNet();
    }
}

File: RetrofitTest/app/src/main/java/contract/FourthContract.kt
package contract

import android.widget.TextView

/**
 * created by shi on 2018/7/6/006
 *
 */
interface FourthContract {
    interface Model {

        fun loginNet(userName: String, pwd: String)
    }

    interface View {

        fun getUserName():String? //可返回null
        fun getPassWord():String //不可返回null ,否则会报错
        fun loginSuccess(success: String)
        fun error(error: String)

    }

    interface Presenter {

        //登录联网
        fun loginNet()
    }

}